hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -423,6 +423,16 @@ private void submitConnectToClusterTask() {
                         submitConnectToClusterTask();
                     }
                 }
+                if (!activeConnections.isEmpty()) {
```

Comment:
I don't think this will work as intended. This method is called after reconnects as well

Reply:
Right, I added a boolean parameter to make this work as expected.